engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -711,6 +711,7 @@ private void handleCreateVolumeFromSnapshotBothOnStorageSystem(SnapshotInfo snap
         }
         catch (Exception ex) {
             errMsg = ex.getMessage() != null ? ex.getMessage() : "Copy operation failed in 'StorageSystemDataMotionStrategy.handleCreateVolumeFromSnapshotBothOnStorageSystem'";
+            copyCmdAnswer = new CopyCmdAnswer(errMsg);
```

Comment:
That would be great @mike-tutkowski. That makes sure that the snapshot goes back to a good state if the storage driver operation fails. I will close this request then.